### DISCUSSION TITLE
WSL: Ubuntu Shortcut Issues

### DISCUSSION THREAD

**Discussion by Niferu** [Cyfrin/foundry-full-course-cu#295] [Q&A] [answered]
Hello everyone,

I'm having hard time figuring out how to use standard commands like `ctrl+shift+left arrow key` in bash terminal. I have installed ubuntu on my windows machine and those are working perfectly fine everywhere (powershell for example), but for bash terminal using `ctrl+shift+left arrow key` just write `D` into terminal awhile using right arrow will write `C`. I couldnt find any solution on google and have found some solutions from chatGPT, but I'm afraid this will completely destroy my terminal short cuts as it is ingerating into default settings. I have also found following topic [link](https://stackoverflow.com/questions/59293850/text-selection-not-works-on-ubuntu-terminal-with-shiftarrow-left-or-right) on stack, but it seems there is no simple solution...

@cromewar 
Sorry to calling you, but it seems you are WSL expert, is there any simple solution to fix those shortcuts as it is really annyoing for me?

**Comment by PatrickAlphaC** [comment]
You'll probably just need to find the new keyboard shortcuts... Idk what those are... Yes maybe cromewar knows.

**Comment by cromewar** [comment]
Hey @Neftyr 
Well Unix based Os works differently from windows, and If you use `ctrl+shift+left arrow` with also type a `D` on the terminal, this is the default behavior for `bash`, `zsh` and `fish`. So I would actually ask what is the behavior you expect? on windows what does that shortcut does? If you hand me that information I would be able to help you finding the equivalent.

**Reply by Niferu** [reply]
Hi @cromewar 

I just expect it to mark whole word. 

For example:

`|` (position of our cursor)`

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/110537615/069be153-d35a-4341-9f0e-87e5a328ab80)

after clicking `ctrl+shift+left arrow`

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/110537615/01a81884-2206-41fe-8a59-291255d41d38)

after clicking `ctrl+shift+left arrow` 3 times

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/110537615/5b238fbd-149c-47e7-a3d9-c75007044ef4)

**Reply by Niferu** [reply]
Any updates on this @cromewar ? :D

**Comment by Equious** [comment, marked as answer]
@Nlferu   You're going to want to look at this, it provides a list of really helpful keyboard shortcuts for the bash terminal.

https://gist.github.com/michaelcoyote/745d8a6cf293c4bf7e31

**Reply by Niferu** [reply]
Hi @Equious 

That is indeed helpful but unfortunately there are no select commands at all. I have tried on both mode vi and emacs nothing works, I believe only way to do this is just to remap default commands.
